openapi/src/apis/actix_server.rs

```diff
@@ -0,0 +1,203 @@
+use actix_web::{http::StatusCode, web::ServiceConfig, FromRequest, HttpResponse, ResponseError};
+use serde::Serialize;
+use std::{
+    fmt::{self, Debug, Display, Formatter},
+    ops,
+};
+
+pub use crate::apis::{
+    block_devices_api::actix::server::BlockDevices, children_api::actix::server::Children,
+    json_grpc_api::actix::server::JsonGrpc, nexuses_api::actix::server::Nexuses,
+    nodes_api::actix::server::Nodes, pools_api::actix::server::Pools,
+    replicas_api::actix::server::Replicas, specs_api::actix::server::Specs,
+    volumes_api::actix::server::Volumes, watches_api::actix::server::Watches,
+};
+
+/// Rest Error wrapper with a status code and a JSON error
+/// Note: Only a single error type for each handler is supported at the moment
+pub struct RestError<T: Debug + Serialize> {
+    status_code: StatusCode,
+    error_response: T,
+}
+
+impl<T: Debug + Serialize> RestError<T> {
+    pub fn new(status_code: StatusCode, error_response: T) -> Self {
+        Self {
+            status_code,
+            error_response,
+        }
+    }
+}
+
+impl<T: Debug + Serialize> Debug for RestError<T> {
+    fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {
+        f.debug_struct("RestError")
+            .field("status_code", &self.status_code)
+            .field("error_response", &self.error_response)
+            .finish()
+    }
+}
+
+impl<T: Debug + Serialize> Display for RestError<T> {
+    fn fmt(&self, _: &mut Formatter<'_>) -> fmt::Result {
+        unimplemented!()
```

Comment:
this needs to be fixed, otherwise any print of the error might crash.

Reply:
Done